app/main/views/suppliers.py

```diff
@@ -408,13 +409,16 @@ def update_supplier_framework_details(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
+    uniform_now = datetime.utcnow()
+
     if 'onFramework' in update_json:
         interest_record.on_framework = update_json['onFramework']
-    if 'agreementReturned' in update_json:
-        if update_json['agreementReturned']:
-            interest_record.agreement_returned_at = datetime.utcnow()
-        else:
-            interest_record.agreement_returned_at = None
+    if update_json.get('agreementReturned'):
+        interest_record.agreement_returned_at = uniform_now
+    if update_json.get('countersigned'):
+        interest_record.countersigned_at = uniform_now
+    if 'signerDetails' in update_json:
+        interest_record.signer_details = update_json["signerDetails"]
```

Comment:
I'm not sure that we should ever need to remove a key from this field - because you can't un-sign a thing.  

But yes, I think we should probably add that validator, so that we get uniform behaviour across the API (i.e. send a null value for a key to delete it from the data).